### PR TITLE
Filter outliers

### PR DESCRIPTION
**Pull request by ellisrichardj**
A number of samples appear to sit on much longer branches than others in the clade and are thus thought to be outliers on the basis of lower quality sequence data.  This PR removes a selection of pre-defined samples from clade datasets from the list provided in outliers.txt.  Interestingly, removing these ~200 samples reduces the run time of the process significantly.  Some increase in resolution (slight increase in branch lengths and small SNP differences between previously identical samples) has been observed as a result.

### CODE REVIEW

**Review by Hira-Naveed — APPROVED** [review]
Hi Richard, many thanks for providing the detailed information on what you have changed in btb-forestry pipeline.  It is really great to see that the removal of these outliers has increased resolution and run time. Great stuff!

**Comment by ellisrichardj** [comment]
> I would usually avoid editing files in-place with sed but happy to accept these changes to reduce the number of files written to disk.

As the file `sed` is reading is from a symlink, it actually doesn't change the original file, but rather creates a new local file with the same name - thus breaking the symlink.  If in-place editing isn't used, because sed is actually iterating over the same file for each entry in outliers.txt and new file will only contain the change from the last iteration.

**Review by nick-pestell — COMMENTED** [review]
Looks good to me, one small comment re. an additional argument on line 16 of the nextflow. 

A few more general comments:
- I feel it might make more sense to have the removing outliers in a separate process, e.g. after splitclades or after sampleLists. That seems more intuitive to me since cladesnps seems to be more geared towards downloading consensus files and building the phylogeny than filtering the samples.
- It would be good to get some unit tests and some CI/CD setup for this pipeline soon. :)

File: main.nf [hunk 1]
```diff
@@ -13,7 +13,7 @@ process cleandata {
         path ('bTB_Allclean_*.csv')
     """
     addsub.sh concat.csv
-    cleanNuniq.sh withsub.csv ${params.today}
+    cleanNuniq.sh withsub.csv ${params.today} outliers.txt
```
[the remark this answers is not in the document]
Comment (ellisrichardj) [reply main.nf]: Good spot - left over from a previous iteration - now removed

**Comment by ellisrichardj** [comment]
> Looks good to me, one small comment re. an additional argument on line 16 of the nextflow.
> 
> A few more general comments:
> 
> * I feel it might make more sense to have the removing outliers in a separate process, e.g. after splitclades or after sampleLists. That seems more intuitive to me since cladesnps seems to be more geared towards downloading consensus files and building the phylogeny than filtering the samples.

I had been undecided about where this should sit - and have moved it around.  I have now opted for changing `sampleLists` to `filterSamples` and moving all the filtering (Ncount threshold and removing outliers) to this new process.

> * It would be good to get some unit tests and some CI/CD setup for this pipeline soon. :)

Indeed!!

**Review by nick-pestell — APPROVED** [review]
Looks good. Thanks for making those changes!

**Pull request merged**